xcape/core.py

```diff
@@ -129,40 +129,78 @@ def calc_cape(p, t, td, ps, ts, tds, **kwargs):
     zMUlev : array-like
         height of MUlev (m)
      """
-    
+    if len(args)>1:
+        print('too many optional arguments')
+        
+    elif kwargs['vertical_lev']!=True:
+        print('please select type of vertical grid, pressure level, or model level')
+        
+    elif (len(args)==0)&(kwargs['vertical_lev'] == 'pressure'):
+        print('vertical grid is set to pressure levels,\n'+
+                +'but location in fortran values (1: nlev) of where p <= ps is missing')
+        
```

Comment:
if there is no optional argument (pres_lev_pos) but vertical_lev==pressure, then it should stop and throw an error

Reply:
raise, don't print